File: WinForms/VB/TreeListMainDemo/frmMain.vb
Imports DevExpress.DXperience.Demos
Imports DevExpress.LookAndFeel
Imports DevExpress.Tutorials
Imports DevExpress.XtraEditors
Imports DevExpress.XtraNavBar

Namespace DevExpress.XtraTreeList.Demos

    Public Class frmMain
        Inherits DXperience.Demos.RibbonMainForm

        Protected Overrides Sub SetFormParam()
            Icon = Utils.ResourceImageHelperCore.CreateIconFromResourcesEx("DevExpress.XtraTreeList.Demos.AppIcon.ico", GetType(frmMain).Assembly)
        End Sub

        Protected Overrides ReadOnly Property DemoName As String
            Get
                Return "WinForms TreeList"
            End Get
        End Property

        Protected Overrides ReadOnly Property FileType As SourceFileType
            Get
                Return DemoHelper.GetSourceFileType()
            End Get
        End Property

        Protected Overrides ReadOnly Property ProductName As String
            Get
                Return "XtraTreeList"
            End Get
        End Property

        Protected Overrides Sub ShowAbout()
            Call TreeList.About()
        End Sub

        Protected Overrides ReadOnly Property SupportAdvancedTitlePainting As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property UseEmptyRootPadding As Boolean
            Get
                Return True
            End Get
        End Property

        Protected Overrides Function CreateRibbonMenuManager() As RibbonMenuManager
            Return New TreeListRibbonMenuManager(Me)
        End Function

        Protected Overrides Sub ShowModule(ByVal name As String, ByVal group As GroupControl, ByVal lookAndFeel As DefaultLookAndFeel)
            Call DemosInfo.ShowModule(name, group, TryCast(RibbonMenuManager, TreeListRibbonMenuManager))
        End Sub

        Protected Overrides Sub FillNavBar()
            ModulesInfo.FillAccordionControl(accordionControl1, NavBarGroupStyle.SmallIconsText, True, NavBarImage.Large)
        End Sub
    End Class
End Namespace

File: WinForms/CS/TreeListMainDemo/frmMain.cs
using DevExpress.DXperience.Demos;
using DevExpress.LookAndFeel;
using DevExpress.Tutorials;
using DevExpress.XtraEditors;
using DevExpress.XtraNavBar;

namespace DevExpress.XtraTreeList.Demos {
    public class frmMain : DevExpress.DXperience.Demos.RibbonMainForm {
        protected override void SetFormParam() {
            this.Icon = DevExpress.Utils.ResourceImageHelper.CreateIconFromResourcesEx("DevExpress.XtraTreeList.Demos.AppIcon.ico", typeof(frmMain).Assembly);
        }
        protected override string DemoName {
            get { return "WinForms TreeList"; }
        }
        protected override SourceFileType FileType {
            get { return DemoHelper.GetSourceFileType(); }
        }
        protected override string ProductName {
            get { return "XtraTreeList"; }
        }
        protected override void ShowAbout() {
            TreeList.About();
        }
        protected override bool SupportAdvancedTitlePainting {
            get { return false; }
        }
        public override bool UseEmptyRootPadding {
            get { return true; }
        }
        protected override RibbonMenuManager CreateRibbonMenuManager() {
            return new TreeListRibbonMenuManager(this);
        }
        protected override void ShowModule(string name, GroupControl group, DefaultLookAndFeel lookAndFeel) {
            DemosInfo.ShowModule(name, group, RibbonMenuManager as TreeListRibbonMenuManager);
        }
        protected override void FillNavBar() {
            DemosInfo.FillAccordionControl(accordionControl1, NavBarGroupStyle.SmallIconsText, true, NavBarImage.Large);
        }
    }
}
